src/libraries/System.Console/src/System/ConsolePal.Android.cs

```diff
@@ -0,0 +1,168 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal sealed unsafe class LogcatStream : ConsoleStream
+    {
+        public LogcatStream() : base(FileAccess.Write) {}
+
+        public override int Read(byte[] buffer, int offset, int count) => throw Error.GetReadNotSupported();
+
+        public override unsafe void Write(byte[] buffer, int offset, int count)
+        {
+            string log = ConsolePal.OutputEncoding.GetString(buffer, offset, count);
+            Interop.Logcat.LogInfo(null, log);
+        }
+    }
+
+    internal static class ConsolePal
+    {
+        internal static void EnsureConsoleInitialized() { }
+
+        public static Stream OpenStandardInput() => throw new PlatformNotSupportedException();
+
+        public static Stream OpenStandardOutput() => new LogcatStream();
+
+        public static Stream OpenStandardError() => new LogcatStream();
+
+        public static Encoding InputEncoding => throw new PlatformNotSupportedException();
+
+        public static void SetConsoleInputEncoding(Encoding enc) => throw new PlatformNotSupportedException();
+
+        public static Encoding OutputEncoding => Encoding.Unicode;
```

Comment:
Just checking, it's Unicode rather than UTF8?

Reply:
`Encoding.Unicode` just tells the Console to dump `System.String` "as is" without unicode->utf8 conversion to ` void Write(byte[] buffer, int offset, int count)` where I call that logcat method and convert byte-array back to string and let the marshaller do the conversion to utf8 with `\0` in the end. The workflow can be improved to reduce a couple of allocations, not sure it worth doing now.